### ISSUE TITLE
DL3018 incorrectly warns on virtual Alpine package addition

### ISSUE THREAD

**Issue by sanmai-NL** [hadolint/hadolint#906]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

For DL3018, Hadolint only parses package expressions, and skips other arguments.

### Actual behavior

```text
Dockerfile:4 DL3018 warning: Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`
```

### Steps to reproduce the behavior

```sh
hadolint Dockerfile
```

I suspect the `buildonly` argument of the `--virtual` parameter was incorrectly parsed as a package expression.

Suggestion: enforce termination of the argument list with `--`, if that makes for easier parsing, given the maintenance burden of possibly changing parameters to `apk add`.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```text
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1
FROM alpine:3.16 AS production
RUN apk \
  add \
  --no-cache \
  --virtual buildonly
  git~=2.38
```

**Additional environment details (OS, stack version, etc.)**
